### PR TITLE
[refactor-push][m]: refactored push function to update status.csv - r…

### PR DESCRIPTION
**Pull request by Mikanebu**
* Refactored `CoreTools.push` method to check `status.csv` file for column `validated == true` and if so it pushes and updates column `published = true`
* Plus, added test

### CODE REVIEW

File: test/index.test.js [hunk 1]
```diff
@@ -1,21 +1,65 @@
...
+const nock = require('nock')
+const sinon = require('sinon')
+const {DataHub} = require('datahub-cli/dist/utils/datahub.js')
+const {Package} = require('datahub-cli/dist/utils/data.js')
 
 const {CoreTools} = require('../index.js')
 
 const statusCsv = path.join(__dirname, 'status-test.csv')
 
+nock('https://api-test.com')
```
Comment (anuveyatsu) [inline test/index.test.js]: We are not using `nock` anymore so can you remove this code from line 12 to 33.

File: test/index.test.js [hunk 2]
```diff
@@ -1,21 +1,65 @@
 const path = require('path')
 const test = require('ava')
+const nock = require('nock')
```
Comment (anuveyatsu) [inline test/index.test.js]: Also remove importing `nock` here

File: test/index.test.js (hunk 1 above)
Comment (Mikanebu) [reply test/index.test.js]: ok

File: package.json [hunk 3]
```diff
@@ -39,7 +39,9 @@
     "datahub-cli": "git+https://github.com/datahq/datahub-cli.git",
     "json2csv": "^3.10.0",
     "lodash": "^4.17.4",
+    "nock": "^9.0.14",
```
Comment (anuveyatsu) [inline package.json]: and please delete `nock` and update `package.json`

**Review by anuveyatsu — CHANGES_REQUESTED** [review]
See comments above

File: package.json (hunk 3 above)
Comment (Mikanebu) [reply package.json]: ok

**Review by anuveyatsu — APPROVED** [review]
LGMT

**Pull request merged**